### ISSUE TITLE
Brain_Data() loading slow when data does not have standard FOV

### ISSUE THREAD

**Issue by ljchang** [ljchang/nltools#27]
This is due to slow automatic resampling to the standard image space.  Need to decide whether to turn this off by default to speed up loading.

**Comment by ljchang** [comment]
Removed automatic resampling line as this was redundant with nifti_masker.fit_transform.  Now we need to decide if we always want to transform all data to brain_mask, or if we want to create a brain_mask for each dataset as default.  Probably faster, but will potentially create more problems comparing data or calculating similarity later.

**Comment by ljchang** [comment]
@burnash curious about your opinion about this.  Right now every time you load data using the Brain_Data() class it makes sure that the data is transformed into a standard 2 x 2 x 2 space with the same field of view.  This is good for ensuring that all methods will work on all data (e.g., plotting, and similarity), but bad because it dramatically slows down loading big datasets. On my tests turning this off can result in loading files about 4 times faster.  On the other hand if we turn this off and allow the class to create an implicit mask for each datafile, I would need to build tests to resample anyway every time it needs to operate on a standard image space.  I'm inclined to leave it how it is for now, but it's possible we could find a way to make calculating the spatial similarity much faster for the website if we need to.

**Comment by burnash** [comment]
@ljchang To leave it as it is — do you mean no resampling by default? 
I checked the Brain_Data definition and it looks like the [`resample`](https://github.com/ljchang/neurolearn/blob/9e98e457c3761b2a20c17b2cf3d764ddcd55a63e/nltools/data.py#L38) argument is not being used at the moment. Maybe use it as an explicit flag to turn resampling on or off?

**Comment by ljchang** [comment]
I don't think we need resample at all because we are converting to the brain mask image by default
```
self.nifti_masker = NiftiMasker(mask_img=self.mask)
self.data = self.nifti_masker.fit_transform(data)
```
My question was more about should we even do this by default?  We could instead create an implicit mask for each nibabel instance and then use this to convert to a Brain_Data() instance. We could then set an explicit flag to use the brain_mask if we want, but there wouldn't be any resampling done by default.  The advantage is that nibabel is very fast and we would be able to work with the data instantly.  The disadvantage is that we would eventually need to convert data to a common space when we do things like calculate similarity, so it might not actually buy us much time savings.

**Comment by ljchang** [comment]
I think this might be better addressed now in PR #149
